### PR TITLE
Added support for permit2 scheme

### PR DESCRIPTION
**Pull request by EvgenKor**
Updated with the latest changes of the client library to support new deposit schemes. CLI changes:
- old method to perform deposit via approval `deposit-shielded` was removed
- `deposit-shielded-permittable` was renamed to `deposit-shielded`
- `deposit-shielded-permittable-ephemeral` was renamed to `deposit-shielded-ephemeral`

Associated library PR: https://github.com/zkBob/zkbob-client-js/pull/140

### CODE REVIEW

File: client-config.json [hunk 1]
```diff
@@ -18,15 +19,28 @@
             "relayerUrls": ["https://dev-relayer.thgkjlr.website/"],
             "delegatedProverUrls": [],
             "coldStorageConfigPath": "",
-            "feeDecimals": 1
+            "feeDecimals": 2,
+            "depositScheme": "permit"
+        },
+        "WETH-goerli": {
```
Comment (k1rill-fedoseev) [inline client-config.json:25]: Should we also add a USDC pool on goerli in this PR?

File: src/account.ts [hunk 2]
```diff
@@ -534,90 +536,51 @@ export class Account {
...
+                    await this.getClient().increaseAllowance(this.getTokenAddr(), this.getPoolAddr(), totalNeededAmount.toString());
+                } else {
+                    console.log(`Current allowance (${await this.weiToHuman(currentAllowance)} ${this.tokenSymbol()}) is greater or equal than needed (${await this.weiToHuman(totalNeededAmount)} ${this.tokenSymbol()}). Skipping approve`);
+                }
+            } else if (depositScheme == DepositType.PermitV2) {
+                const currentAllowance = await this.getClient().allowance(this.getTokenAddr(), PERMIT2_CONTRACT);
+                if (totalNeededAmount > currentAllowance) {
+                    const maxTokensAmount = 2n ** 256n - 1n;
+                    console.log(`Approving Permit2 contract (${PERMIT2_CONTRACT}) to spend max amount of our tokens`);
+                    await this.getClient().approve(this.getTokenAddr(), PERMIT2_CONTRACT, maxTokensAmount.toString());
```
Comment (r0wdy1) [inline src/account.ts:567]: isn't it better to make unlimited allowance? 
https://docs.uniswap.org/contracts/permit2/overview#approving-permit2

File: client-config.json (hunk 1 above)
Comment (EvgenKor) [reply client-config.json:25]: Sure, added in https://github.com/zkBob/zkbob-console/pull/85/commits/e659e1d63810e1a5885a9208218d6a82aa2a481b. You can try it with `USDC-goerli` alias

**Review by k1rill-fedoseev — APPROVED** [review]

File: src/account.ts [hunk 3]
```diff
@@ -370,13 +373,15 @@ export class Account {
...
+        const tokenAddress = this.config.pools[this.getCurrentPool()].tokenAddress;
+        return await this.getClient().fromBaseUnit(tokenAddress, amountWei.toString());
     }
 
 
     public async getBalance(): Promise<[string, string]> {
+        const tokenAddress = this.config.pools[this.getCurrentPool()].tokenAddress;
         const balance = await this.getClient().getBalance();
-        const readable = this.getClient().fromBaseUnit(balance);
+        const readable = await this.getClient().fromBaseUnit(tokenAddress, balance);
```
Comment (AllFi) [inline src/account.ts]: It seems that we shouldn't use `tokenAddress` here. 
image
image

**Review by AllFi — APPROVED** [review]
Apart from the bug in the `get-balance` function, everything looks good to me.

File: src/account.ts (hunk 3 above)
Comment (EvgenKor) [reply src/account.ts]: Oh thanks! Fixed in https://github.com/zkBob/zkbob-console/pull/85/commits/2cde248b397e35f50b8c03c52c4c5249e1f09c8f

Comment (EvgenKor) [reply src/account.ts]: And also found another bugs with the same conversion https://github.com/zkBob/zkbob-console/pull/85/commits/6c4b78b47157a4706c40cd054120111e6d39388c

**Pull request merged**